prowler/providers/aws/services/kms/kms_service.py

```diff
@@ -43,73 +48,93 @@ def _list_keys(self, regional_client):
-                key.spec = response["KeyMetadata"]["CustomerMasterKeySpec"]
-                key.multi_region = response["KeyMetadata"]["MultiRegion"]
-            except Exception as error:
-                logger.error(
-                    f"{regional_client.region} -- {error.__class__.__name__}:{error.__traceback__.tb_lineno} -- {error}"
-                )
-
-    def _get_key_rotation_status(self):
-        logger.info("KMS - Get Key Rotation Status...")
-        for key in self.keys:
-            if (
-                key.origin
-                and key.manager
-                and "EXTERNAL" not in key.origin
-                and "AWS" not in key.manager
-            ):
+        try:
+            for key in self.keys:
                 regional_client = self.regional_clients[key.region]
                 try:
-                    key.rotation_enabled = regional_client.get_key_rotation_status(
-                        KeyId=key.id
-                    )["KeyRotationEnabled"]
+                    response = regional_client.describe_key(KeyId=key.id)
+                    key.state = response["KeyMetadata"]["KeyState"]
+                    key.origin = response["KeyMetadata"]["Origin"]
+                    key.manager = response["KeyMetadata"]["KeyManager"]
+                    key.spec = response["KeyMetadata"]["CustomerMasterKeySpec"]
+                    key.multi_region = response["KeyMetadata"]["MultiRegion"]
                 except Exception as error:
                     logger.error(
                         f"{regional_client.region} -- {error.__class__.__name__}:{error.__traceback__.tb_lineno} -- {error}"
                     )
+        except Exception as error:
+            logger.error(
+                f"{regional_client.region} -- {error.__class__.__name__}:{error.__traceback__.tb_lineno} -- {error}"
+            )
+
+    def _get_key_rotation_status(self):
+        logger.info("KMS - Get Key Rotation Status...")
+        try:
+            for key in self.keys:
+                if (
+                    key.origin
+                    and key.manager
+                    and "EXTERNAL" not in key.origin
+                    and "AWS" not in key.manager
+                ):
+                    regional_client = self.regional_clients[key.region]
+                    try:
+                        key.rotation_enabled = regional_client.get_key_rotation_status(
+                            KeyId=key.id
+                        )["KeyRotationEnabled"]
+                    except Exception as error:
+                        logger.error(
+                            f"{regional_client.region} -- {error.__class__.__name__}:{error.__traceback__.tb_lineno} -- {error}"
+                        )
+        except Exception as error:
+            logger.error(
+                f"{regional_client.region} -- {error.__class__.__name__}:{error.__traceback__.tb_lineno} -- {error}"
+            )
```

Comment:
@MrCloudSec the outer `try-except` is redundant, right?

Reply:
No, since we are covering any potential errors in the lines 73-80